### DISCUSSION TITLE
Multiple status codes for the same endpoint

### DISCUSSION THREAD

**Discussion by adel-chouchane** [microcks/microcks#809] [Q&A]
Hello,
I noticed that by default when you define multiple status codes for your endpoint Microcks returns randomly each time a status code.
Is there a way to force a specific status code response. I asked chat gpt and this was it's answer :

> Yes, you can force Microcks to return a specific response for an API endpoint during testing by using the "request headers" feature in Microcks.
To force Microcks to return a specific response, you can add a custom header to your API request with the name "X-Microcks-Response-Code" and the value of the desired HTTP status code. For example, to force a 404 response for a GET /users endpoint, you would add the following header to your API request:
X-Microcks-Response-Code: 404
When Microcks receives this request, it will override the default behavior and return the specified HTTP status code and response defined in your OpenAPI definition file.
Note that this feature should be used for testing purposes only and should not be relied upon in production environments. Additionally, be sure to remove any custom headers added to your API requests before deploying your API to production.

But it's not working, is there a real solution to this ?
Thanks in advance.

**Comment by lbroudoux** [comment]
Hi @adel-chouchane 

I'm very surprised of the above quoted block as I cannot identify this being part of official documentation of an answer someone has given on GitHub or Zulip... Could you share the origin of this quote? 

That said, mocking  and dispatching to multiple status code should work in a very deterministic way. Depending on the dispatcher configured on your operation, Microcks is trying to find an appropriate response matching the dispatch criteria. Once this response is found, it is using the response code (200, 201, 404, ...) attached to this response. As an example, you can check this [blog post](https://microcks.io/blog/advanced-dispatching-constraints/) where the example returns `404` when asking weather forecast for an unknown region.

Let us know if that helps 😉

**Reply by adel-chouchane** [reply]
For the quoted text as I said it's from chat GPT.

**Reply by lbroudoux** [reply]
ha ha funny. Didn't see it at first. Looks like Chat GPT creates additional features 😉

**Comment by adel-chouchane** [comment]
Hello again, @lbroudoux , 
Is there a way to determine status code response for a GET list, for a GET by ID or a request with query params it's easy to do it via dispatchers, but for a GET on a list for example GET /api/cars (list all cars), is there a way you recommend to have the 200 response or other status code response ?
Thanks in advance

**Reply by lbroudoux** [reply]
Hello!
As the default inferred dispatcher when no input is random, you can use a customized dispatcher using either a `SCRIPT` or a `FALLBACK` dispatcher. More on that here: https://microcks.io/documentation/using/dispatching/
